src/main/java/io/r2dbc/h2/codecs/BytesCodec.java

```diff
@@ -38,6 +38,6 @@ boolean doCanDecode(int dataType) {
 
     @Override
     Value doEncode(byte[] value) {
-        return ValueBytes.get(Assert.requireNonNull(value, "value must not be null"));
+        return ValueBinary.get(Assert.requireNonNull(value, "value must not be null"));
```

Comment:
When a binary string literal is specified, its data type `BINARY` or `BINARY VARYING` is implementation-defined in the SQL Standard. H2 uses `BINARY VARYING` and usage of `BINARY` may cause various unexpected effects.

So here `ValueVarbinary.get()` should be used for consistency.

Reply:
👍